### DISCUSSION TITLE
Open Label in a new Window

### DISCUSSION THREAD

**Discussion by Pixeltrace-GER** [inventree/InvenTree#3594] [Q&A] [answered]
Hey,
I'm in love with InvenTree.  I don't know why I didn't discover this nice tool, earlier.

At the moment, I'm setting up all the Stock items. Printing the Label with my Dymo 450 works fine. 
But here are my questions I'm looking into for two days.

Is it possible to open the PDF Label in a new Window and not in the same as the modal? 
Is there a possibility to start the printing dialog automatically?

Many thanks for all the work on InvenTree <3

**Comment by matmair** [comment, marked as answer]
@Pixeltrace-GER there are [printing plugins](https://inventree.readthedocs.io/en/latest/extend/integrate/) for brother and zebra printers,  you could probably write one for dynamo.

**Reply by Pixeltrace-GER** [reply]
I can check it. Maybe I'm able to write one. 
But I was hoping there was a little option to open the PDF-Label in a new window. 
Thank you for the quick replay.

**Reply by Pixeltrace-GER** [reply]
I've checked some files.  To open the Label in a new Window, I need to change the `printLabels` function in *.labels.js.

`window.location.href = url; ` in `window.open( url , '_blank');`

I'm not a good programmer, so I don't know how much work it is, to add a Toggle-Button in the (Admin)Interface.
For me, I will change, this afternoon, the static files. I home this is a quick solution.

**Reply by matmair** [reply]
@Pixeltrace-GER there is a PR implementing this.
In general, I do not recommend changing the code locally - you should really update regularly and will need to reapply your changes every time (and be careful that nothing around it changed).

**Reply by SchrodingersGat** [reply]
https://github.com/inventree/InvenTree/pull/3602

**Reply by Pixeltrace-GER** [reply]
@matmair With editing the static files, I only want to test if it is possible. But in this case, I will take a look at the PR. I will do update regularly because I hosting it on a VPS.

@SchrodingersGat Thank you for linking the PR

You are awesome, guys. I think this will be my solution.
